src/components/Comments.jsx

```diff
@@ -22,72 +25,94 @@ export default function Comments({ id }) {
-  const [isDeleteComplete, setIsDeleteComplete] = useState({
+
+  const isUsersComment = user === author;
+
+  return (
+    <div className="comment-card">
+      <p>
+        {moment(created_at).format('MMMM Do YYYY')} -{' '}
+        <Link to={`/users/${author}`} className="article-card-link">
+          u/{author}
+        </Link>
+      </p>
+      <p>{body}</p>
+      <div>
+        {isUsersComment && (
+          <DisplayButton
+            comment_id={comment_id}
+            setUserHasDeletedComment={setUserHasDeletedComment}
+          />
+        )}
+      </div>
+    </div>
+  );
+}
+
+function DisplayButton({ comment_id, setUserHasDeletedComment }) {
+  const [deleteStatus, setDeleteStatus] = useState({
     loading: false,
     complete: false,
   });
-  const isUsersComment = user === author;
-
-  const deleteComment = () => {
-    setIsDeleteComplete(() => {
-      return { ...isDeleteComplete, loading: true };
-    });
 
-    deleteArticleComment(comment_id).then(() => {
-      setIsDeleteComplete({ loading: false, complete: true });
-    });
-  };
+  useEffect(() => {
+    if (deleteStatus.complete) {
+      setUserHasDeletedComment(true);
+    } else setUserHasDeletedComment(false);
+  }, [deleteStatus.complete]);
 
-  function displayButton() {
-    // Planned re-factor as this is overly excessive, <Button> tag does not need to be repeated.
-    if (!isDeleteComplete.loading && !isDeleteComplete.complete) {
-      return (
-        <Button variant="outline" size="sm-icon" onClick={deleteComment}>
-          <Trash2 className="h-4 w-4" />
-        </Button>
-      );
-    }
+  let displayedButton = null;
+  let disabled = false;
 
-    if (isDeleteComplete.loading && !isDeleteComplete.complete) {
-      return (
-        <Button
-          variant="outline"
-          size="sm-icon"
-          onClick={deleteComment}
-          disabled
-        >
-          {<Loader2 className="h-4 w-4 animate-spin" />}
-        </Button>
-      );
-    }
+  if (!deleteStatus.loading && !deleteStatus.complete) {
+    displayedButton = <Trash2 className="h-4 w-4" />;
+  } else if (deleteStatus.loading && !deleteStatus.complete) {
+    displayedButton = <Loader2 className="h-4 w-4 animate-spin" />;
+    disabled = true;
+  } else if (!deleteStatus.loading && deleteStatus.complete) {
+    displayedButton = <CircleCheck className="h-4 w-4" />;
+  }
+  const deleteComment = () => {
+    setDeleteStatus({ loading: true, complete: false });
 
-    if (!isDeleteComplete.loading && isDeleteComplete.complete) {
-      return (
-        <Button variant="outline" size="sm-icon">
-          <CircleCheck className="h-4 w-4" />
-        </Button>
+    deleteArticleComment(comment_id)
+      .then(() => {
+        setDeleteStatus({ loading: false, complete: true });
+      })
+      .then(() =>
+        setTimeout(() => {
+          setDeleteStatus({ loading: false, complete: false });
```

Comment:
Here im trying to re-render after the delete. setTimeout doesn't feel optimal here so please suggest how this could be done in a proper way.

Reply:
I see your point here. A better way would be to catch an error from the delete request to the db and respond to that.

Reply:
An error isn't thrown - this is more about updating the cards so does not trigger a whole page refresh